pkg/controllers/core/deployer_controller.go

```diff
@@ -616,5 +675,18 @@ func (r *DeployerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Watches(&source.Kind{Type: &buildv1alpha1.Application{}}, enqueueTrackedResources(&buildv1alpha1.Application{})).
 		Watches(&source.Kind{Type: &buildv1alpha1.Container{}}, enqueueTrackedResources(&buildv1alpha1.Container{})).
 		Watches(&source.Kind{Type: &buildv1alpha1.Function{}}, enqueueTrackedResources(&buildv1alpha1.Function{})).
+		// watch for service mutations to update colliding deployers
+		Watches(&source.Kind{Type: &corev1.Service{}}, enqueueResourcesCollidingWithServices()).
```

Comment:
`Owns` is a more specific form of `Watches`, since Service is already "owned" this isn't necessary.

Reply:
Interesting. I thought that `Owns` would only track services which are owned by deployers. The reason for using `Watches` too was to track colliding services which are not owned by deployers. I know you don't want to track "unowned" services, but if we did decide to do that, would `Owns` achive it?